### PR TITLE
[Serve] Support customizable readiness probe timeout

### PR DESCRIPTION
**Pull request by cblmemo**
Expose the readiness probe timeout to users. Example:

```yaml
# test.yaml
service:
  readiness_probe:
    path: /health
    initial_delay_seconds: 20
    timeout_seconds: 100
  replicas: 1

resources:
  cpus: 2+
  ports: 8081

run: python -m http.server 8081
```

```bash
$ sky serve up test.yaml
Service from YAML spec: test.yaml
Service Spec:
Readiness probe method:           GET /health
Readiness initial delay seconds:  20
Readiness probe timeout seconds:  100
Replica autoscaling policy:       Fixed 1 replica
Spot Policy:                      No spot policy
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
     - [x] The YAML above
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: new smoke tests added in this PR 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/serve/replica_managers.py
Comment (Michaelvll) [inline sky/serve/replica_managers.py:496]: Should we use the timeout for the post request directly or should we instead handle the timeout ourselves for when to set the replica to be `NOT_READY`? Reason: changing the timeout for the post request will cause the probing blocked by the a single request.

Comment (cblmemo) [reply sky/serve/replica_managers.py:496]: Could you elaborate on `handle the timeout ourselves for when to set the replica to be NOT_READY`..? Do you refer to letting the user configure the consecutive failure timeout?

Comment (cblmemo) [reply sky/serve/replica_managers.py:496]: Though it is from a user's request

**Comment by Michaelvll** [comment]
Hey @cblmemo, I merged the master branch and made some minor changes to this PR, so our user could try it out. After thinking more, the discussion above might be fine, as long as we explicitly mention that a too-large timeout can cause issues in detecting failures. Let's try to get the PR in soon. : )

**Comment by cblmemo** [comment]
> Hey @cblmemo, I merged the master branch and made some minor changes to this PR, so our user could try it out. After thinking more, the discussion above might be fine, as long as we explicitly mention that a too-large timeout can cause issues in detecting failures. Let's try to get the PR in soon. : )

Thanks for updating this! It looks great to me and is ready to go. Though seems like I cannot approve my own PR, so I'll probably merge this after you approve this ;)

**Review by Michaelvll — APPROVED** [review]
The PR looks good to me @cblmemo! Could you help add and run some smoke tests before we merge this PR?

**Comment by cblmemo** [comment]
All newly-added smoke test passed. Merging now 🫡

**Pull request merged**
